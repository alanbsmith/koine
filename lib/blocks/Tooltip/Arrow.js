import styled from 'styled-components';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

const modifierConfig = {
  bottom: (props) => `
    border-top-width: ${rem(props.theme.dimensions.basePad)};
    border-right-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-width: 0;
    border-left-width: ${rem(props.theme.dimensions.basePad)};
    border-top-color: ${props.theme.colors.base.chrome900};
    bottom: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
  left: (props) => `
    border-top-width: ${rem(props.theme.dimensions.basePad)};
    border-right-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-width: ${rem(props.theme.dimensions.basePad)};
    border-left-width: 0;
    border-right-color: ${props.theme.colors.base.chrome900};
    left: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
  right: (props) => `
    border-top-width: ${rem(props.theme.dimensions.basePad)};
    border-right-width: 0;
    border-bottom-width: ${rem(props.theme.dimensions.basePad)};
    border-left-width: ${rem(props.theme.dimensions.basePad)};
    border-left-color: ${props.theme.colors.base.chrome900};
    right: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
  top: (props) => `
    border-top-width: 0;
    border-right-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-width: ${rem(props.theme.dimensions.basePad)};
    border-left-width: ${rem(props.theme.dimensions.basePad)};
    border-bottom-color: ${props.theme.colors.base.chrome900};
    top: ${rem(-props.theme.dimensions.basePad + 1)};
  `,
};

const Arrow = styled.div`
  border-color: transparent;
  border-style: solid;
  height: 0;
  position: absolute;
  width: 0;
  ${applyStyleModifiers(modifierConfig)}
`;

export default Arrow;
