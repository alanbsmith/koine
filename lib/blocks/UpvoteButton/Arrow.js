import React from 'react';
import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';
import { rem } from 'polished';
import FontAwesome from 'react-fontawesome';

function UnstyledArrow(props) {
  return (
    <FontAwesome
      name="arrow-up"
      {...props}
    />
  );
};

const modifierConfig = {
  orange: ({ theme }) => `color: ${theme.colors.base.orange};`,
  hoverBounce: ({ theme }) => `
    &:hover {
      animation: ${theme.animations.upvote} 1s infinite ease-in;
    }
  `,
};

const Arrow = styled(UnstyledArrow)`
  font-size: ${rem(24)};
  ${applyStyleModifiers(modifierConfig)}
`;

export default Arrow;
