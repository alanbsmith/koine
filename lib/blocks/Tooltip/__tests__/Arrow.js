import React from 'react';
import Arrow from '../Arrow';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('Tooltip Arrow Element', () => {
  it('applies the bottom modifier', () => {
    const tree = renderWithTheme(
      <Arrow modifiers={['bottom']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies the left modifier', () => {
    const tree = renderWithTheme(
      <Arrow modifiers={['left']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies the right modifier', () => {
    const tree = renderWithTheme(
      <Arrow modifiers={['right']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies the top modifier', () => {
    const tree = renderWithTheme(
      <Arrow modifiers={['top']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
