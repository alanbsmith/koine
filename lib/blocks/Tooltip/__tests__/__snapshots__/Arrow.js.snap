// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Arrow Element applies the bottom modifier 1`] = `
.c0 {
  border-color: transparent;
  border-style: solid;
  height: 0;
  position: absolute;
  width: 0;
  border-top-width: 0.5rem;
  border-right-width: 0.5rem;
  border-bottom-width: 0;
  border-left-width: 0.5rem;
  border-top-color: #191919;
  bottom: -0.4375rem;
}

<div
  className="c0"
/>
`;

exports[`Tooltip Arrow Element applies the left modifier 1`] = `
.c0 {
  border-color: transparent;
  border-style: solid;
  height: 0;
  position: absolute;
  width: 0;
  border-top-width: 0.5rem;
  border-right-width: 0.5rem;
  border-bottom-width: 0.5rem;
  border-left-width: 0;
  border-right-color: #191919;
  left: -0.4375rem;
}

<div
  className="c0"
/>
`;

exports[`Tooltip Arrow Element applies the right modifier 1`] = `
.c0 {
  border-color: transparent;
  border-style: solid;
  height: 0;
  position: absolute;
  width: 0;
  border-top-width: 0.5rem;
  border-right-width: 0;
  border-bottom-width: 0.5rem;
  border-left-width: 0.5rem;
  border-left-color: #191919;
  right: -0.4375rem;
}

<div
  className="c0"
/>
`;

exports[`Tooltip Arrow Element applies the top modifier 1`] = `
.c0 {
  border-color: transparent;
  border-style: solid;
  height: 0;
  position: absolute;
  width: 0;
  border-top-width: 0;
  border-right-width: 0.5rem;
  border-bottom-width: 0.5rem;
  border-left-width: 0.5rem;
  border-bottom-color: #191919;
  top: -0.4375rem;
}

<div
  className="c0"
/>
`;
